apis/v1alpha1/shared_types.go

```diff
@@ -80,77 +96,111 @@ type Port struct {
+	//
 	// +optional
-	Namespaces *NamespacedPeer `json:"namespaces,omitempty"`
+	Namespaces *metav1.LabelSelector `json:"namespaces,omitempty"`
 	// Pods defines a way to select a set of pods in
-	// in a set of namespaces.
+	// a set of namespaces. Note that host-networked pods
+	// are not included in this type of peer.
+	//
+	// Support: Core
+	//
 	// +optional
-	Pods *NamespacedPodPeer `json:"pods,omitempty"`
+	Pods *NamespacedPod `json:"pods,omitempty"`
 }
 
-// NamespacedPeer defines a flexible way to select Namespaces in a cluster.
-// Exactly one of the selectors must be set.  If a consumer observes none of
-// its fields are set, they must assume an unknown option has been specified
-// and fail closed.
+// AdminNetworkPolicyEgressPeer defines a peer to allow traffic to.
+// Exactly one of the selector pointers must be set for a given peer. If a
+// consumer observes none of its fields are set, they must assume an unknown
+// option has been specified and fail closed.
 // +kubebuilder:validation:MaxProperties=1
 // +kubebuilder:validation:MinProperties=1
-type NamespacedPeer struct {
-	// NamespaceSelector is a labelSelector used to select Namespaces, This field
-	// follows standard label selector semantics; if present but empty, it selects
-	// all Namespaces.
+type AdminNetworkPolicyEgressPeer struct {
+	// Namespaces defines a way to select all pods within a set of Namespaces.
+	// Note that host-networked pods are not included in this type of peer.
+	//
+	// Support: Core
+	//
 	// +optional
-	NamespaceSelector *metav1.LabelSelector `json:"namespaceSelector,omitempty"`
-
-	// SameLabels is used to select a set of Namespaces that share the same values
-	// for a set of labels.
-	// To be selected a Namespace must have all of the labels defined in SameLabels,
-	// AND they must all have the same value as the subject of this policy.
-	// If Samelabels is Empty then nothing is selected.
+	Namespaces *metav1.LabelSelector `json:"namespaces,omitempty"`
+	// Pods defines a way to select a set of pods in
+	// a set of namespaces. Note that host-networked pods
+	// are not included in this type of peer.
+	//
+	// Support: Core
+	//
+	// +optional
+	Pods *NamespacedPod `json:"pods,omitempty"`
+	// Nodes defines a way to select a set of nodes in
+	// the cluster. This field follows standard label selector
+	// semantics; if present but empty, it selects all Nodes.
+	//
+	// Support: Extended
+	//
+	// <network-policy-api:experimental>
 	// +optional
-	// +kubebuilder:validation:MaxItems=100
-	SameLabels []string `json:"sameLabels,omitempty"`
-
-	// NotSameLabels is used to select a set of Namespaces that do not have certain
-	// values for a set of label(s).
-	// To be selected a Namespace must have all of the labels defined in NotSameLabels,
-	// AND at least one of them must have different values than the subject of this policy.
-	// If NotSameLabels is empty then nothing is selected.
+	Nodes *metav1.LabelSelector `json:"nodes,omitempty"`
```

Comment:
ok, the diff is ugly here, but the API change is the addition of `Nodes` to `AdminNetworkPolicyEgressPeer`.

The docs are inadequate; what does selecting Nodes mean? In particular, we had requests in the past for both "select the IPs of nodes" and "select all pod-network pods on a node".

I believe it means the former, but what _exactly_ does "the IPs of nodes" mean? The primary IPs? All NodeAddresses? All IPs assigned to an interface in the host network namespace on the node, whether or not they are listed in any k8s.io API object?

Reply:
@danwinship opened a PR to address this: https://github.com/kubernetes-sigs/network-policy-api/pull/280